In [92]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly
import plotly.express as px

In [93]:
dataset = pd.read_csv('dataset/serl_energy_use_in_GB_domestic_buildings_2021_aggregated_statistics_v01.csv')

/var/folders/lw/cbxh4d5n6nj91c7dnbv6yy740000gn/T/ipykernel_13408/2668219060.py:1: DtypeWarning:

Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.



In [94]:
dataset['time_period'].replace(2020, '2020', inplace=True)
dataset['time_period'].replace(2021, '2021', inplace=True)

In [95]:
origin_dataset = dataset
dataset.head()

fuel unit        summary_stat subsample summary_time  \
0  Net Electricity  kWh                mean       all         2021   
1  Net Electricity  kWh  standard deviation       all         2021   
2  Net Electricity  kWh              median       all         2021   
3  Net Electricity  kWh     75th percentile       all         2021   
4  Net Electricity  kWh     25th percentile       all         2021   

  time_period segmentation_variable_1 segment_1_value   value  n_sample  ...  \
0        2021                    None            None   9.823   10764.0  ...   
1        2021                    None            None   7.700   10764.0  ...   
2        2021                    None            None   8.168   10764.0  ...   
3        2021                    None            None  12.212   10764.0  ...   
4        2021                    None            None   5.415   10764.0  ...   

   mean_temp  mean_hdd  decimal_places  weekday_weekend mean_floor_area  \
0     10.378      5.74             3.0             both       95.551547   
1     10.378      5.74             3.0             both       95.551547   
2     10.378      5.74             3.0             both       95.551547   
3     10.378      5.74             3.0             both       95.551547   
4     10.378      5.74             3.0             both       95.551547   

   n_mean_floor_area  mean_bedrooms  n_mean_bedrooms  mean_occupants  \
0             7617.0       2.918951          12659.0        2.277386   
1             7617.0       2.918951          12659.0        2.277386   
2             7617.0       2.918951          12659.0        2.277386   
3             7617.0       2.918951          12659.0        2.277386   
4             7617.0       2.918951          12659.0        2.277386   

   n_mean_occupants  
0           12625.0  
1           12625.0  
2           12625.0  
3           12625.0  
4           12625.0  

[5 rows x 21 columns]

In [96]:
def monthly_common_data(dataframe, summary_stat):
    return dataframe.loc[(dataframe['summary_stat'] == summary_stat)
                       & (dataframe['segmentation_variable_1'] == 'None')
                       & (dataframe['summary_time'] != '2021')
                       & (dataframe['summary_time'] != '2020')
                       & (dataframe['summary_time'] != 'heating_season_2019_2020')
                       & (dataframe['summary_time'] != 'heating_season_2020_2021')
                       & (dataframe['mean_hdd'])      # a little bit trick here to remove the every 30mins summary time.
    ]

temp = monthly_common_data(dataset, 'mean')
temp

fuel unit summary_stat subsample summary_time time_period  \
20    Net Electricity  kWh         mean       all       Jan-21        2021   
25                Gas  kWh         mean       all       Jan-21        2021   
30    Net Electricity  kWh         mean       all       Feb-21        2021   
35                Gas  kWh         mean       all       Feb-21        2021   
40    Net Electricity  kWh         mean       all       Mar-21        2021   
45                Gas  kWh         mean       all       Mar-21        2021   
50    Net Electricity  kWh         mean       all       Apr-21        2021   
55                Gas  kWh         mean       all       Apr-21        2021   
60    Net Electricity  kWh         mean       all       May-21        2021   
65                Gas  kWh         mean       all       May-21        2021   
70    Net Electricity  kWh         mean       all       Jun-21        2021   
75                Gas  kWh         mean       all       Jun-21        2021   
80    Net Electricity  kWh         mean       all       Jul-21        2021   
85                Gas  kWh         mean       all       Jul-21        2021   
90    Net Electricity  kWh         mean       all       Aug-21        2021   
95                Gas  kWh         mean       all       Aug-21        2021   
100   Net Electricity  kWh         mean       all       Sep-21        2021   
105               Gas  kWh         mean       all       Sep-21        2021   
110   Net Electricity  kWh         mean       all       Oct-21        2021   
115               Gas  kWh         mean       all       Oct-21        2021   
120   Net Electricity  kWh         mean       all       Nov-21        2021   
125               Gas  kWh         mean       all       Nov-21        2021   
130   Net Electricity  kWh         mean       all       Dec-21        2021   
135               Gas  kWh         mean       all       Dec-21        2021   
1120  Net Electricity  kWh         mean       all       Jan-20        2020   
1125              Gas  kWh         mean       all       Jan-20        2020   
1130  Net Electricity  kWh         mean       all       Feb-20        2020   
1135              Gas  kWh         mean       all       Feb-20        2020   
1140  Net Electricity  kWh         mean       all       Mar-20        2020   
1145              Gas  kWh         mean       all       Mar-20        2020   
1150  Net Electricity  kWh         mean       all       Apr-20        2020   
1155              Gas  kWh         mean       all       Apr-20        2020   
1160  Net Electricity  kWh         mean       all       May-20        2020   
1165              Gas  kWh         mean       all       May-20        2020   
1170  Net Electricity  kWh         mean       all       Jun-20        2020   
1175              Gas  kWh         mean       all       Jun-20        2020   
1180  Net Electricity  kWh         mean       all       Jul-20        2020   
1185              Gas  kWh         mean       all       Jul-20        2020   
1190  Net Electricity  kWh         mean       all       Aug-20        2020   
1195              Gas  kWh         mean       all       Aug-20        2020   
1200  Net Electricity  kWh         mean       all       Sep-20        2020   
1205              Gas  kWh         mean       all       Sep-20        2020   
1210  Net Electricity  kWh         mean       all       Oct-20        2020   
1215              Gas  kWh         mean       all       Oct-20        2020   
1220  Net Electricity  kWh         mean       all       Nov-20        2020   
1225              Gas  kWh         mean       all       Nov-20        2020   
1230  Net Electricity  kWh         mean       all       Dec-20        2020   
1235              Gas  kWh         mean       all       Dec-20        2020   

     segmentation_variable_1 segment_1_value   value  n_sample  ...  \
20                      None            None  13.862   11979.0  ...   
25                      None            None  80.868

In [97]:
def summary_time_year_remove(dataframe):
    for i in range(dataframe.shape[0]):
        dataframe.iloc[i,4] = dataframe.iloc[i,4][:-3]
    return dataframe

summary_time_year_remove(temp)

fuel unit summary_stat subsample summary_time time_period  \
20    Net Electricity  kWh         mean       all          Jan        2021   
25                Gas  kWh         mean       all          Jan        2021   
30    Net Electricity  kWh         mean       all          Feb        2021   
35                Gas  kWh         mean       all          Feb        2021   
40    Net Electricity  kWh         mean       all          Mar        2021   
45                Gas  kWh         mean       all          Mar        2021   
50    Net Electricity  kWh         mean       all          Apr        2021   
55                Gas  kWh         mean       all          Apr        2021   
60    Net Electricity  kWh         mean       all          May        2021   
65                Gas  kWh         mean       all          May        2021   
70    Net Electricity  kWh         mean       all          Jun        2021   
75                Gas  kWh         mean       all          Jun        2021   
80    Net Electricity  kWh         mean       all          Jul        2021   
85                Gas  kWh         mean       all          Jul        2021   
90    Net Electricity  kWh         mean       all          Aug        2021   
95                Gas  kWh         mean       all          Aug        2021   
100   Net Electricity  kWh         mean       all          Sep        2021   
105               Gas  kWh         mean       all          Sep        2021   
110   Net Electricity  kWh         mean       all          Oct        2021   
115               Gas  kWh         mean       all          Oct        2021   
120   Net Electricity  kWh         mean       all          Nov        2021   
125               Gas  kWh         mean       all          Nov        2021   
130   Net Electricity  kWh         mean       all          Dec        2021   
135               Gas  kWh         mean       all          Dec        2021   
1120  Net Electricity  kWh         mean       all          Jan        2020   
1125              Gas  kWh         mean       all          Jan        2020   
1130  Net Electricity  kWh         mean       all          Feb        2020   
1135              Gas  kWh         mean       all          Feb        2020   
1140  Net Electricity  kWh         mean       all          Mar        2020   
1145              Gas  kWh         mean       all          Mar        2020   
1150  Net Electricity  kWh         mean       all          Apr        2020   
1155              Gas  kWh         mean       all          Apr        2020   
1160  Net Electricity  kWh         mean       all          May        2020   
1165              Gas  kWh         mean       all          May        2020   
1170  Net Electricity  kWh         mean       all          Jun        2020   
1175              Gas  kWh         mean       all          Jun        2020   
1180  Net Electricity  kWh         mean       all          Jul        2020   
1185              Gas  kWh         mean       all          Jul        2020   
1190  Net Electricity  kWh         mean       all          Aug        2020   
1195              Gas  kWh         mean       all          Aug        2020   
1200  Net Electricity  kWh         mean       all          Sep        2020   
1205              Gas  kWh         mean       all          Sep        2020   
1210  Net Electricity  kWh         mean       all          Oct        2020   
1215              Gas  kWh         mean       all          Oct        2020   
1220  Net Electricity  kWh         mean       all          Nov        2020   
1225              Gas  kWh         mean       all          Nov        2020   
1230  Net Electricity  kWh         mean       all          Dec        2020   
1235              Gas  kWh         mean       all          Dec        2020   

     segmentation_variable_1 segment_1_value   value  n_sample  ...  \
20                      None            None  13.862   11979.0  ...   
25                      None            None  80.868

In [98]:
def select_fuel(dataframe, fuel):
    return dataframe.loc[dataframe['fuel'] == fuel]

temp = select_fuel(dataframe=temp, fuel='Net Electricity')
temp

fuel unit summary_stat subsample summary_time time_period  \
20    Net Electricity  kWh         mean       all          Jan        2021   
30    Net Electricity  kWh         mean       all          Feb        2021   
40    Net Electricity  kWh         mean       all          Mar        2021   
50    Net Electricity  kWh         mean       all          Apr        2021   
60    Net Electricity  kWh         mean       all          May        2021   
70    Net Electricity  kWh         mean       all          Jun        2021   
80    Net Electricity  kWh         mean       all          Jul        2021   
90    Net Electricity  kWh         mean       all          Aug        2021   
100   Net Electricity  kWh         mean       all          Sep        2021   
110   Net Electricity  kWh         mean       all          Oct        2021   
120   Net Electricity  kWh         mean       all          Nov        2021   
130   Net Electricity  kWh         mean       all          Dec        2021   
1120  Net Electricity  kWh         mean       all          Jan        2020   
1130  Net Electricity  kWh         mean       all          Feb        2020   
1140  Net Electricity  kWh         mean       all          Mar        2020   
1150  Net Electricity  kWh         mean       all          Apr        2020   
1160  Net Electricity  kWh         mean       all          May        2020   
1170  Net Electricity  kWh         mean       all          Jun        2020   
1180  Net Electricity  kWh         mean       all          Jul        2020   
1190  Net Electricity  kWh         mean       all          Aug        2020   
1200  Net Electricity  kWh         mean       all          Sep        2020   
1210  Net Electricity  kWh         mean       all          Oct        2020   
1220  Net Electricity  kWh         mean       all          Nov        2020   
1230  Net Electricity  kWh         mean       all          Dec        2020   

     segmentation_variable_1 segment_1_value   value  n_sample  ...  \
20                      None            None  13.862   11979.0  ...   
30                      None            None  12.987   12037.0  ...   
40                      None            None  11.484   11951.0  ...   
50                      None            None  10.110   11840.0  ...   
60                      None            None   9.129   11797.0  ...   
70                      None            None   7.512   11803.0  ...   
80                      None            None   7.451   11825.0  ...   
90                      None            None   7.575   11605.0  ...   
100                     None            None   7.692   11663.0  ...   
110                     None            None   8.903   11774.0  ...   
120                     None            None  10.396   11547.0  ...   
130                     None            None  11.540   11683.0  ...   
1120                    None            None  12.227    4050.0  ...   
1130                    None            None  11.858    7324.0  ...   
1140                    None            None  11.438    8627.0  ...   
1150                    None            None  10.141    8744.0  ...   
1160                    None            None   9.192    9459.0  ...   
1170                    None            None   8.920    9528.0  ...   
1180                    None            None   8.577    9790.0  ...   
1190                    None            None   8.468   10235.0  ...   
1200                    None            None   8.809   10769.0  ...   
1210                    None            None  10.517   11164.0  ...   
1220                    None            None  11.718   11419.0  ...   
1230                    None            None  13.231   11871.0  ...   

      mean_temp  mean_hdd  decimal_places  weekday_weekend mean_floor_area  \
20        3.583    12.037             3.0             both       95.551547   
30        5.185    10.376             3.0             both       95.551547   
40        7.048     8.466             3.0             bo

In [99]:
dataset.loc[(dataset['segmentation_variable_1'] == 'Region')
            & (dataset['summary_stat'] == 'mean')
            & (dataset['summary_time'] == '2021')]

fuel unit summary_stat subsample summary_time time_period  \
76010  Net Electricity  kWh         mean       all         2021        2021   
76015              Gas  kWh         mean       all         2021        2021   
77230  Net Electricity  kWh         mean       all         2021        2021   
77235              Gas  kWh         mean       all         2021        2021   
78450  Net Electricity  kWh         mean       all         2021        2021   
78455              Gas  kWh         mean       all         2021        2021   
79670  Net Electricity  kWh         mean       all         2021        2021   
79675              Gas  kWh         mean       all         2021        2021   
80890  Net Electricity  kWh         mean       all         2021        2021   
80895              Gas  kWh         mean       all         2021        2021   
82110  Net Electricity  kWh         mean       all         2021        2021   
82115              Gas  kWh         mean       all         2021        2021   
83330  Net Electricity  kWh         mean       all         2021        2021   
83335              Gas  kWh         mean       all         2021        2021   
84550  Net Electricity  kWh         mean       all         2021        2021   
84555              Gas  kWh         mean       all         2021        2021   
85770  Net Electricity  kWh         mean       all         2021        2021   
85775              Gas  kWh         mean       all         2021        2021   
86990  Net Electricity  kWh         mean       all         2021        2021   
86995              Gas  kWh         mean       all         2021        2021   
88210  Net Electricity  kWh         mean       all         2021        2021   
88215              Gas  kWh         mean       all         2021        2021   

      segmentation_variable_1  segment_1_value   value  n_sample  ...  \
76010                  Region    WEST MIDLANDS   9.254     954.0  ...   
76015                  Region    WEST MIDLANDS  38.079     771.0  ...   
77230                  Region    EAST MIDLANDS   9.505     762.0  ...   
77235                  Region    EAST MIDLANDS  39.779     657.0  ...   
78450                  Region       SOUTH WEST   9.657     974.0  ...   
78455                  Region       SOUTH WEST  33.112     692.0  ...   
79670                  Region   GREATER LONDON  10.291    1403.0  ...   
79675                  Region   GREATER LONDON  41.931     941.0  ...   
80890                  Region       NORTH EAST   8.876     396.0  ...   
80895                  Region       NORTH EAST  37.421     301.0  ...   
82110                  Region       SOUTH EAST  10.523    1425.0  ...   
82115                  Region       SOUTH EAST  40.572    1068.0  ...   
83330                  Region         SCOTLAND   9.838    1010.0  ...   
83335                  Region         SCOTLAND  38.743     574.0  ...   
84550                  Region       NORTH WEST   9.702    1312.0  ...   
84555                  Region       NORTH WEST  38.983    1043.0  ...   
85770                  Region  EAST OF ENGLAND  10.351     969.0  ...   
85775                  Region  EAST OF ENGLAND  39.377     718.0  ...   
86990                  Region            WALES   9.223     655.0  ...   
86995                  Region            WALES  35.432     506.0  ...   
88210                  Region        YORKSHIRE   9.480     904.0  ...   
88215                  Region        YORKSHIRE  38.193     691.0  ...   

       mean_temp  mean_hdd  decimal_places  weekday_weekend mean_floor_area  \
76010     10.382     5.814             3.0             both       92.125174   
76015     10.378     5.817             3.0             both       92.125174   
77230     10.350     5.815             3.0             both       92.367041   
77235     10.343     5.822             3.0             both       92.367041   
78450     11.063     5.036             3.0             both       99.568821   
78455     11.079     5.017             3.0    

In [107]:
def get_pv_annual_data(dataframe, fuel, summary_stat):
    pv_annual_data = dataframe.loc[(dataframe['segmentation_variable_1'] == 'has_pv')
            & (dataframe['summary_stat'] == summary_stat)
            & (dataframe['fuel'] == fuel)
            & (dataframe['summary_time'] != 'heating_season_2019_2020')
            & (dataframe['summary_time'] != 'heating_season_2020_2021')
            & ((dataframe['summary_time'] == '2020') | (dataframe['summary_time'] == '2021'))
            & (dataframe['mean_hdd'])]
    return pv_annual_data

def get_pv_heating_season_data(dataframe, fuel, summary_stat):
    pv_heating_season_data = dataframe.loc[(dataframe['segmentation_variable_1'] == 'has_pv')
            & (dataframe['summary_stat'] == summary_stat)
            & (dataframe['fuel'] == fuel)
            & ((dataframe['summary_time'] == 'heating_season_2019_2020')
            | (dataframe['summary_time'] == 'heating_season_2020_2021'))
            & (dataframe['summary_time'] != '2020')
            & (dataframe['summary_time'] != '2021')
            & (dataframe['mean_hdd'])]
    return pv_heating_season_data

def get_pv_monthly_data(dataframe, fuel, summary_stat):
    pv_monthly_data = dataframe.loc[(dataframe['segmentation_variable_1'] == 'has_pv')
            & (dataframe['summary_stat'] == summary_stat)
            & (dataframe['fuel'] == fuel)
            & (dataframe['summary_time'] != 'heating_season_2019_2020')
            & (dataframe['summary_time'] != 'heating_season_2020_2021')
            & (dataframe['summary_time'] != '2020')
            & (dataframe['summary_time'] != '2021')
            & (dataframe['mean_hdd'])]
    return pv_monthly_data

get_pv_annual_data(dataset, fuel='Net Electricity', summary_stat='mean')

fuel unit summary_stat subsample summary_time time_period  \
69910  Net Electricity  kWh         mean       all         2021        2021   
70520  Net Electricity  kWh         mean       all         2020        2020   
71130  Net Electricity  kWh         mean       all         2021        2021   
71740  Net Electricity  kWh         mean       all         2020        2020   

      segmentation_variable_1 segment_1_value   value  n_sample  ...  \
69910                  has_pv           False  10.081   10168.0  ...   
70520                  has_pv           False  10.909    3614.0  ...   
71130                  has_pv            True   5.417     596.0  ...   
71740                  has_pv            True   5.915      90.0  ...   

       mean_temp  mean_hdd  decimal_places  weekday_weekend mean_floor_area  \
69910     10.376     5.743             3.0             both       93.954067   
70520     11.025     5.160             3.0             both       93.954067   
71130     10.420     5.686             3.0             both      115.005969   
71740     11.359     4.889             3.0             both      115.005969   

       n_mean_floor_area  mean_bedrooms  n_mean_bedrooms  mean_occupants  \
69910             7039.0       2.893843          11888.0        2.276499   
70520             7039.0       2.893843          11888.0        2.276499   
71130              578.0       3.306096            771.0        2.291123   
71740              578.0       3.306096            771.0        2.291123   

       n_mean_occupants  
69910           11859.0  
70520           11859.0  
71130             766.0  
71740             766.0  

[4 rows x 21 columns]

In [101]:
fig_pv_monthly_gas = px.line(summary_time_year_remove(get_pv_monthly_data(dataset, fuel='Gas', summary_stat='mean')),
               x="summary_time", y="value", color="segment_1_value",
              labels={
                  'summary_time': 'Month',
                  'value': 'Consumption',
                  'time_period': 'year',
                  'segment_1_value': 'has_pv'
              },
              animation_frame="time_period", animation_group="value")

fig_pv_monthly_gas.update_layout(xaxis=dict(showgrid=False))
fig_pv_monthly_gas.show()

This is the plot for the gas consumption group by *'has_pv'* segmentation.
We found that the pattern of the mean gas consumption in each month was similar, which means that **the use of gas do not effected by whether the building has pv or not.**

In [102]:
fig_pv_monthly_electricity = px.line(summary_time_year_remove(get_pv_monthly_data(dataset, fuel='Net Electricity', summary_stat='mean')),
                x="summary_time", y="value", color="segment_1_value",
                labels={
                    'summary_time': 'Month',
                    'value': 'Consumption',
                    'time_period': 'year',
                    'segment_1_value': 'has_pv'
                },
                animation_frame="time_period", animation_group="value")

fig_pv_monthly_electricity.update_layout(xaxis=dict(showgrid=False))
fig_pv_monthly_electricity.show()

This is the plot for the electricity consumption group by *‘has_pv’* segmentation. We found that **those households with pv(Photovoltaics) used less electricity on average per household in both 2020 and 2021, and in most months, they have less electricity consumption on average per household.**
We also found that the pattern of electricity consumption of those who have no pv is quite similar as monthly the gas consumption pattern, higher in colder month and lower in Jun to Sep.
But to those who have pv, the pattern is different. They had much lower consumption of electricity in Apr to Sep, especially in Jun and July, the electricity their pv produce can cover the usage. Their electricity consumption pattern were more significantly influenced by time. This is mainly caused by the sunshine duration and solar elevation angle.
So, the pv is a very important factor which impact the energy use pattern and can help a lot with decrease the usage of net electricity.
If pv can be widely used, our need of net electricity supply will decrease, which also help low the stress on electricity generation and transmission.

In [103]:
# electricity_value_per_m2 = (pv_monthly_electricity['value']/pv_monthly_electricity['mean_floor_area']).tolist()
# pv_monthly_electricity_value_per_m2 = pv_monthly_electricity.loc[:,
#                                       ['fuel', 'unit', 'summary_stat', 'summary_time','time_period', 'segment_1_value', 'value', 'n_sample', 'mean_floor_area', 'n_mean_floor_area']]
# pv_monthly_electricity_value_per_m2['value_per_m2'] = electricity_value_per_m2
# pv_monthly_electricity_value_per_m2

def get_value_per_m2_data(dataframe):
    value_per_m2 = (dataframe['value']/dataframe['mean_floor_area']).tolist()
    value_per_m2_data = dataframe.loc[:,
                                          ['fuel', 'unit', 'summary_stat', 'summary_time','time_period', 'segment_1_value', 'value', 'n_sample', 'mean_floor_area', 'n_mean_floor_area']]
    value_per_m2_data['value_per_m2'] = value_per_m2
    return value_per_m2_data


pv_monthly_electricity_data = summary_time_year_remove(get_pv_monthly_data(dataset, fuel='Net Electricity', summary_stat='mean'))
pv_monthly_gas_data = summary_time_year_remove(get_pv_monthly_data(dataset, fuel='Gas', summary_stat='mean'))


# get_value_per_m2_data(pv_monthly_electricity_data)
# get_value_per_m2_data(pv_monthly_gas_data)

In [104]:
fig_pv_monthly_electricity_value_per_m2 = px.line(get_value_per_m2_data(pv_monthly_electricity_data),
               x="summary_time", y="value_per_m2", color="segment_1_value",
               title=(pv_monthly_electricity_data['fuel'].unique()[0]) + ' Consumption per m^2',
               labels={
                   'summary_time': 'Month',
                   'value': 'Consumption',
                   'time_period': 'year',
                   'segment_1_value': 'has_pv'
               },
               animation_frame="time_period", animation_group="value_per_m2")

fig_pv_monthly_electricity_value_per_m2.update_layout(xaxis=dict(showgrid=False))
fig_pv_monthly_electricity_value_per_m2.show()

In [106]:
fig_pv_monthly_gas_value_per_m2 = px.line(get_value_per_m2_data(pv_monthly_gas_data),
               x="summary_time", y="value_per_m2", color="segment_1_value",
               title=(pv_monthly_gas_data['fuel'].unique()[0]) + ' Consumption per m^2',
               labels={
                   'summary_time': 'Month',
                   'value': 'Consumption',
                   'time_period': 'year',
                   'segment_1_value': 'has_pv'
               },
               animation_frame="time_period", animation_group="value_per_m2")

fig_pv_monthly_gas_value_per_m2.update_layout(xaxis=dict(showgrid=False))
fig_pv_monthly_gas_value_per_m2.show()

Only analyze the value of different types of fuel group by whether has pv or not can cause some bias: we can easily found that, **the households which have pv also have much bigger mean household area (no pv:93.954067,have pv:115.005969), that is very likely because the richer the family, the larger the household area and they are more likely to afford the expense of installation of pv.**
And in the relevance analysis, we found area is a highly correlated variable. So we want to eliminate the effect of the area to the fuel value. So we think the **fuel consumption per m2** can be a better way to do deeper explore with the data.
Then we found something amazing that: **The average fuel consumption per square metre for households with pv in either gas or electricity is lower in any given month.**
That is a good news for all the residents, **pv can help lower their expanse on energy, pay less to energy suppliers and save their money.**